src/context/intermediate/modals/Input.tsx

```diff
@@ -139,6 +140,33 @@ export function SpecialInputModal(props: SpecialProps) {
                 />
             );
         }
+        case "join_server": {
+            return (
+                <InputModal
+                    onClose={onClose}
+                    question={<Text id="app.main.servers.join" />}
+                    field={"Invite code"}
+                    callback={async (rawCode) => {
+                        // if the user provides an invite link, get rid of the url and just pass the invite code
+                        // prettier-ignore
+                        const regex = new RegExp(
+                            "http(s?):\/\/(app|nightly|rvlt|localhost).(revolt.chat|gg\/|\d{3,5})(\/invite\/)?", // localhost doesn't quite work yet
+                        );
+                        const code = rawCode.replace(regex, "");
+                        const server = await client.fetchInvite(code);
+                        if (typeof server === "undefined")
+                            console.log("Something went wrong.");
+
+                        if (client.servers.get(server.server_id)) {
+                            history.push(
+                                `/server/${server.server_id}/channel/${server.channel_id}`,
+                            );
+                        }
+                        await client.joinInvite(code);
```

Comment:
Should probably open the server after joining, although for this the logic from the invite page and embed should be combined and also used here.